File: Python/Algorithms/1299.py
class Solution:
    def replaceElements(self, arr):
        max_, arr[-1] = arr[-1], - 1
        for i in range(len(arr) - 2, -1, -1):
            val, arr[i] = arr[i], max_
            if val > max_:
                max_ = val
        return arr

File: Rust/Algorithms/1299.rs
impl Solution {
    pub fn replace_elements(mut arr: Vec<i32>) -> Vec<i32> {
        let mut max = 0;
        let len = arr.len();
        for i in (0..len).rev() {
            let val = arr[i];
            arr[i] = max;
            if val > max {
                max = val;
            }
        }
        arr[len - 1] = -1;
        arr
    }
}
